nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/xml/WriteXMLResult.java

```diff
@@ -0,0 +1,602 @@
+import org.apache.nifi.serialization.record.RawRecordWriter;
+import org.apache.nifi.serialization.record.Record;
+import org.apache.nifi.serialization.record.RecordField;
+import org.apache.nifi.serialization.record.RecordFieldType;
+import org.apache.nifi.serialization.record.RecordSchema;
+import org.apache.nifi.serialization.record.type.ArrayDataType;
+import org.apache.nifi.serialization.record.type.ChoiceDataType;
+import org.apache.nifi.serialization.record.type.MapDataType;
+import org.apache.nifi.serialization.record.type.RecordDataType;
+import org.apache.nifi.serialization.record.util.DataTypeUtils;
+
+import javax.xml.stream.XMLOutputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamWriter;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.text.DateFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Supplier;
+
+
+public class WriteXMLResult extends AbstractRecordSetWriter implements RecordSetWriter, RawRecordWriter {
+
+    final ComponentLog logger;
+    final RecordSchema recordSchema;
+    final SchemaAccessWriter schemaAccess;
+    final XMLStreamWriter writer;
+    final NullSuppression nullSuppression;
+    final ArrayWrapping arrayWrapping;
+    final String arrayTagName;
+    final String recordTagName;
+    final String rootTagName;
+
+    private final Supplier<DateFormat> LAZY_DATE_FORMAT;
+    private final Supplier<DateFormat> LAZY_TIME_FORMAT;
+    private final Supplier<DateFormat> LAZY_TIMESTAMP_FORMAT;
+
+    public WriteXMLResult(final ComponentLog logger, final RecordSchema recordSchema, final SchemaAccessWriter schemaAccess, final OutputStream out, final boolean prettyPrint,
+                          final NullSuppression nullSuppression, final ArrayWrapping arrayWrapping, final String arrayTagName, final String rootTagName, final String recordTagName,
+                          final String dateFormat, final String timeFormat, final String timestampFormat) throws IOException {
+
+        super(out);
+
+        this.logger = logger;
+        this.recordSchema = recordSchema;
+        this.schemaAccess = schemaAccess;
+        this.nullSuppression = nullSuppression;
+
+        this.arrayWrapping = arrayWrapping;
+        this.arrayTagName = arrayTagName;
+
+        this.rootTagName = rootTagName;
+        this.recordTagName = recordTagName;
+
+        final DateFormat df = dateFormat == null ? null : DataTypeUtils.getDateFormat(dateFormat);
+        final DateFormat tf = timeFormat == null ? null : DataTypeUtils.getDateFormat(timeFormat);
+        final DateFormat tsf = timestampFormat == null ? null : DataTypeUtils.getDateFormat(timestampFormat);
+
+        LAZY_DATE_FORMAT = () -> df;
+        LAZY_TIME_FORMAT = () -> tf;
+        LAZY_TIMESTAMP_FORMAT = () -> tsf;
+
+        try {
+            XMLOutputFactory factory = XMLOutputFactory.newInstance();

```

Comment:
should I consider the encoding somehow (e. g. via property)?
e. g. writer = factory.createXMLStreamWriter(out, StandardCharsets.UTF_8.name());